src/patients/ContactInfo.tsx

```diff
@@ -0,0 +1,211 @@
+    )
+    onChange(newData)
+  }
+
+  const header = (
+    <Row className="header mb-2">
+      <Column xs={12} sm={4}>
+        <span className="">{typeLabel}</span>
+        <span className="d-sm-none"> &amp; {t(label)}</span>
+      </Column>
+      <Column className="d-none d-sm-block" sm={8}>
+        {t(label)}
+      </Column>
+    </Row>
+  )
+
+  // todo: acts strange when deleting empty rows above non-empty rows.
+  // suspect TextInputWithLabelFormGroup missing value
+  const entries = data.map((entry, i) => {
+    const error = getError(i)
+    return (
+      // todo: want a better key, and possibly name
+      // eslint-disable-next-line react/no-array-index-key
+      <Row key={i}>
+        <Column sm={4}>
+          <SelectWithLabelFormGroup
+            name={`${name}Type${i}`}
+            value={entry.type}
+            isEditable={isEditable}
+            options={typeOptions}
+            onChange={
+              onChange
+                ? (event) => {
+                    // eslint-disable-next-line no-shadow
+                    const newData = data.map(({ value, type }) =>
+                      value === entry.value
+                        ? { value, type: event.currentTarget.value }
+                        : { value, type },
+                    )
+                    onChange(newData)
+                  }
+                : undefined
+            }
+          />
+        </Column>
+        <Column sm={8}>
+          {['tel', 'email'].indexOf(type) > -1 ? (
+            <TextInputWithLabelFormGroup
+              name={`${name}${i}`}
+              value={entry.value}
+              isEditable={isEditable}
+              onChange={
+                onChange
+                  ? (event) => {
+                      onChangeValue(event, entry.value)
+                    }
+                  : undefined
+              }
+              feedback={error}
+              isInvalid={!!error}
+              type={type}
+            />
+          ) : (
+            <TextFieldWithLabelFormGroup
+              name={`${name}${i}`}
+              value={entry.value}
+              isEditable={isEditable}
+              onChange={
+                onChange
+                  ? (event) => {
+                      onChangeValue(event, entry.value)
+                    }
+                  : undefined
+              }
+              feedback={error}
+              isInvalid={!!error}
+            />
+          )}
+        </Column>
+      </Row>
+    )
+  })
+
+  const onClickAdd = () => {
+    if (!isValid || !onChange) {
+      return
+    }
+
+    // 1. pick up only non-empty string
+    const newData = data.filter(({ value }) => value.trim() !== '')
+
+    // 2. gather errors
+    const newTempErrors = newData.map(({ value }) => (isValid(value) ? '' : errorMessageLabel))
+
+    // 3. if no error, add a new entry
+    if (newTempErrors.filter((error) => error !== '').length === 0) {
```

Comment:
this can be simplified by doing `newTempErrors.some((e) => e !== ''))`

Reply:
Line no longer exists.